opm/parser/eclipse/RawDeck/RawRecord.hpp

```diff
@@ -34,32 +37,27 @@ namespace Opm {
     public:
         RawRecord(const std::string& singleRecordString, const std::string& fileName = "", const std::string& keywordName = "");
 
-        std::string pop_front();
+        string_view pop_front();
         void push_front(std::string token);
         size_t size() const;
 
         const std::string& getRecordString() const;
-        const std::string& getItem(size_t index) const;
+        string_view getItem(size_t index) const;
         const std::string& getFileName() const;
         const std::string& getKeywordName() const;
 
         static bool isTerminatedRecordString(const std::string& candidateRecordString);
-        virtual ~RawRecord();
-        void dump() const;
+
+       void dump() const;
 
     private:
         std::string m_sanitizedRecordString;
-        std::deque<std::string> m_recordItems;
+        std::deque< string_view > m_recordItems;
+        std::vector< std::string > expanded_items;
```

Comment:
I guess these strings are only to ensure that views which are `push_front()`'ed have something to view? In that case e.g. `string_storage` would be a better name.